command/v6/create_service_broker_command_test.go

```diff
@@ -0,0 +1,204 @@
+					Expect(testUI.Out).To(Say("OK"))
+					Expect(testUI.Err).To(Say("a-warning"))
+					Expect(testUI.Err).To(Say("another-warning"))
+					Expect(executeErr).NotTo(HaveOccurred())
+				})
+			})
+
+			When("creating a service broker is unsuccessful", func() {
+				BeforeEach(func() {
+					fakeActor.CreateServiceBrokerReturns(v2action.ServiceBroker{}, []string{"a-warning", "another-warning"}, errors.New("invalid-broker-name"))
+				})
+
+				It("prints the error and warnings returned by Cloud Controller", func() {
+					Expect(fakeActor.CreateServiceBrokerCallCount()).To(Equal(1))
+
+					Expect(testUI.Out).To(Say("Creating service broker cool-broker as some-user..."))
+					Expect(testUI.Err).To(Say("a-warning"))
+					Expect(testUI.Err).To(Say("another-warning"))
+					Expect(executeErr).To(MatchError("invalid-broker-name"))
+				})
+			})
+		})
+
+		Context("space scoped", func() {
+			BeforeEach(func() {
+				fakeSharedActor.CheckTargetReturns(actionerror.NoSpaceTargetedError{BinaryName: binaryName})
+				cmd.SpaceScoped = true
+			})
+
+			It("returns an error", func() {
+				Expect(fakeActor.CreateServiceBrokerCallCount()).To(Equal(0))
+				Expect(fakeSharedActor.CheckTargetCallCount()).To(Equal(1))
+				Expect(executeErr).To(MatchError(actionerror.NoSpaceTargetedError{BinaryName: binaryName}))
+			})
+		})
+	})
+
+	When("the user is targeting an org and space", func() {
+		BeforeEach(func() {
+			fakeSharedActor.CheckTargetReturns(nil)
+			fakeConfig.CurrentUserReturns(
+				configv3.User{Name: "some-user"},
+				nil)
+			fakeConfig.TargetedOrganizationReturns(configv3.Organization{
+				GUID: "some-org-guid",
+				Name: "some-org",
+			})
+			fakeConfig.TargetedSpaceReturns(configv3.Space{
+				GUID: "some-space-guid",
+				Name: "some-space",
+			})
+			fakeActor.CreateServiceBrokerReturns(v2action.ServiceBroker{}, []string{"a-warning", "another-warning"}, nil)
+		})
+
+		Context("space scoped", func() {
+			BeforeEach(func() {
+				cmd.SpaceScoped = true
+			})
+
+			When("all arguments are passed and are valid", func() {
+				It("registers a space scoped service broker", func() {
+					Expect(fakeActor.CreateServiceBrokerCallCount()).To(Equal(1))
+					serviceBroker, username, password, url, spaceGUID := fakeActor.CreateServiceBrokerArgsForCall(0)
+					Expect(serviceBroker).To(Equal("cool-broker"))
+					Expect(username).To(Equal("admin"))
+					Expect(password).To(Equal("password"))
+					Expect(url).To(Equal("https://broker.com"))
+					Expect(spaceGUID).To(Equal("some-space-guid"))
+
+					Expect(testUI.Out).To(Say("Creating service broker cool-broker in org some-org / space some-space as some-user..."))
+					Expect(testUI.Out).To(Say("OK"))
+					Expect(testUI.Err).To(Say("a-warning"))
+					Expect(testUI.Err).To(Say("another-warning"))
+					Expect(executeErr).NotTo(HaveOccurred())
+				})
+			})
+		})
+
+		Context("not space scoped", func() {
+			BeforeEach(func() {
```

Comment:
The default value of `SpaceScoped` is false, so we shouldn't need to reinitialize it here.